### ISSUE TITLE
Media 2022

### ISSUE THREAD

**Issue by rviscomi** [HTTPArchive/almanac.httparchive.org#2884]
# Media 2022

![Media illustration](https://raw.githubusercontent.com/HTTPArchive/almanac.httparchive.org/main/src/static/images/2019/media/hero_lg.jpg)

If you're interested in contributing to the Media chapter of the 2022 Web Almanac, please reply to this issue and indicate which role or roles best fit your interest and availability: [author](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Authors'-Guide), [reviewer](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Reviewers'-Guide), [analyst](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Analysts'-Guide), and/or [editor](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Editors'-Guide).

## Content team

| Lead | Authors | Reviewers | Analysts | Editors | Coordinator |
| ---- | ------- | --------- | -------- | ------- | ----------- |
| @eeeps  | @eeeps @akshay-ranganath | @nhoizey @yoavweiss | @eeeps @akshay-ranganath | @MichaelLewittes | @siakaramalegos  |

<details>
<summary><b>Expand for more information about each role 👀</b></summary>

- The **[content team lead](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Content-Team-Leads'-Guide)** is the chapter owner and responsible for setting the scope of the chapter and managing contributors' day-to-day progress.
- **[Authors](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Authors'-Guide)** are subject matter experts and lead the content direction for each chapter. Chapters typically have one or two authors. Authors are responsible for planning the outline of the chapter, analyzing stats and trends, and writing the annual report.
- **[Reviewers](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Reviewers'-Guide)** are also subject matter experts and assist authors with technical reviews during the planning, analyzing, and writing phases.
- **[Analysts](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Analysts'-Guide)** are responsible for researching the stats and trends used throughout the Almanac. Analysts work closely with authors and reviewers during the planning phase to give direction on the types of stats that are possible from the dataset, and during the analyzing/writing phases to ensure that the stats are used correctly.
- **[Editors](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Editors'-Guide)** are technical writers who have a penchant for both technical and non-technical content correctness. Editors have a mastery of the English language and work closely with authors to help wordsmith content and ensure that everything fits together as a cohesive unit.
- The **[section coordinator](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Section-Leads'-Guide)** is the overall owner for all chapters within a section like "User Experience" or "Page Content" and helps to keep each chapter on schedule.

_Note: The time commitment for each role varies by the chapter's scope and complexity as well as the number of contributors._

For an overview of how the roles work together at each phase of the project, see the [Chapter Lifecycle](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle) doc.
</details>

### [**0. Form the content team**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#0-create-content-team)
+ [x] **May 1**: The content team has at least one author, reviewer, and analyst

### [**1. Plan content**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#1-plan-content)
+ [x] **May 15** The content team has completed the chapter outline in the [draft doc][~google-doc]

### [**2. Gather data**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#2-gather-data)
+ [x] **June 1**: Analysts have added all necessary [custom metrics](https://github.com/HTTPArchive/custom-metrics/blob/main/README.md) and drafted a PR ([example](https://github.com/HTTPArchive/almanac.httparchive.org/pull/1087)) to track query progress
- **June 1 - 15**: HTTP Archive runs the June crawl

### [**3. Validate results**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#3-validate-results)
+ [x] **August 1**: Analysts have queried all metrics and saved the output to the [results sheet][~google-sheets]

### [**4. Draft content**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#4-draft-content)
+ [x] **September 1**: The content team has written, reviewed, and edited the chapter in the [doc][~google-doc]

### [**5. Publication**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#5-publication)
+ [x] **September 15**: The completed chapter and all required [metadata](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Authors'-Guide#metadata-to-add-at-the-top-of-your-chapters) and [figures](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Figures-Guide) are converted to markdown and submitted to [GitHub][~chapter-markdown]
- **September 26**: Target launch date 🚀

## Chapter resources

Refer to these 2022 Media resources throughout the content creation process:

[📄 Google Docs][~google-doc] for outlining and drafting content
[🔍 SQL files][~sql-files] for committing the queries used during analysis
[📊 Google Sheets][~google-sheets] for saving the results of queries
[📝 Markdown file][~chapter-markdown] for publishing content and managing public metadata
[💬 #web-almanac-media on Slack][~slack-channel] for team coordination

[~google-doc]: https://docs.google.com/document/d/117Roz2RlgXdDl9OiDdkrrd1ahnepFeGzIvTB5S4OY9g/edit?usp=sharing
[~sql-files]: https://github.com/HTTPArchive/almanac.httparchive.org/tree/main/sql/2022/media
[~google-sheets]: https://docs.google.com/spreadsheets/d/1T5oVAVmcH3sM6R-WwH4ksr2jFtPhuLXs3-iXXoABb3E/edit?usp=sharing
[~chapter-markdown]: https://github.com/HTTPArchive/almanac.httparchive.org/tree/main/src/content/en/2022/media.md
[~slack-channel]: https://join.slack.com/t/httparchive/shared_invite/zt-45sgwmnb-eDEatOhqssqNAKxxOSLAaA

**Comment by MichaelLewittes** [comment]
I could help with editing ("Publication"), having worked in media and been a print and digital editor for more than two decades.

**Comment by nhoizey** [comment]
I can be a reviewer on this chapter.

**Comment by eeeps** [comment]
I'd love to be an author, analyst, or even lead again (though if anyone else wants to take that, by all means!)

**Comment by akshay-ranganath** [comment]
I can help in analysis and authoring the chapter.

**Comment by siakaramalegos** [comment]
@eeeps going to put you as lead again for now since no one else has volunteered yet :)

**Comment by jyrkialakuijala** [comment]
I'd be happy to contribute as an co-author of this chapter, or possibly only in the content planning part and helping to scope this chapter. I have experience in image format and codec design -- WebP lossless and JPEG XL, LCP's relation to progressive images, and other work related to faster web sites (Brotli, WOFF2).

**Comment by siakaramalegos** [comment]
@jyrkialakuijala cool! Co-authors tend to contribute to writing. If you're wanting to be more of a guide, then reviewer might be a better role. Reviewers contribute at the outline stage as well as reviewing the written article. Or you could focus on writing in specific content areas, depending on how the content lead wants to organize the content.

On the content lead topic, I apparently jumped the gun, and Rick wants to wait for more volunteers just in case. Ideally, we get new voices each year, but that can be hard. :) So holding off on that though I'd still like to meet with you @eeps to get the ball rolling in the interim (see email).

**Comment by siakaramalegos** [comment]
@eeeps @akshay-ranganath just verifying you've already given your email addresses to @rviscomi to set you up in Big Query so you're not charged

**Comment by rviscomi** [comment]
👋 Hi everyone, please make sure you have access to the [content team doc](https://docs.google.com/document/d/117Roz2RlgXdDl9OiDdkrrd1ahnepFeGzIvTB5S4OY9g/edit?usp=sharing) so we can start making progress on Milestone 1, which is to complete the chapter outline by May 15.

Also FYI we have a [`#web-almanac-media`](https://join.slack.com/t/httparchive/shared_invite/zt-45sgwmnb-eDEatOhqssqNAKxxOSLAaA) channel on Slack where you can coordinate on chapter progress.

@eeeps as the content team lead you'll effectively be the project manager for this chapter. You'll help to make sure that everyone is on the same page and the chapter stays on schedule. I'll hand it off to you to coordinate with the rest of the team on the next milestones.

**Comment by rviscomi** [comment]
🛎️ @nhoizey @eeeps @akshay-ranganath @MichaelLewittes reminder to access the [planning doc](https://docs.google.com/document/d/117Roz2RlgXdDl9OiDdkrrd1ahnepFeGzIvTB5S4OY9g/edit?usp=sharing) and start adding your ideas to the outline. What's new with media on the web this year, or what topics would be interesting to revisit from previous years? We're hoping to have the outline completed by May 15 to give us enough time to add any needed custom metrics to the June crawl before it kicks off. Thanks!

**Comment by MichaelLewittes** [comment]
@nhoizey @eeeps @akshay-ranganath Want to meet on Monday to discuss? I'm on Eastern Time (ET). Let me know what works for everyone.

**Comment by eeeps** [comment]
Let's! Sent an invite. I know the time might not work great for @nhoizey...
Also starting things off with an outline based on last year's queries/chapter, to give us a starting point.

**Comment by nhoizey** [comment]
@eeeps 7pm is sometimes manageable, even with kids! 😅

(Sorry for the late answer, I was on holiday until today.)

**Comment by siakaramalegos** [comment]
@akshay-ranganath @nhoizey just a reminder that the outline is due in 2 days. Please request edit access to the google doc, add your name and email to confirm participation, and provide feedback on the outline that Eric started. This deadline is critical as we only have 2 more weeks afterward to finalize custom metrics. Thanks!

**Comment by eeeps** [comment]
@siakaramalegos we went over the outline as a team in a zoom earlier in the week, and had some discussions about changes. I've just integrated those, and think we can fairly call the outline complete 🎉

**Comment by rviscomi** [comment]
FYI I've added @yoavweiss as a chapter reviewer for his help on https://github.com/HTTPArchive/custom-metrics/pull/30. No expectations of reviewing the content necessarily, but want to make sure we do give credit in the by-lines.

**Comment by yoavweiss** [comment]
Thanks! :)

**Comment by siakaramalegos** [comment]
@eeeps @akshay-ranganath  it looks like the analysis is almost complete - can you give us an idea of % complete and timeline on the rest?

@eeeps  when do you think you can begin the draft? Just as a reminder, the due date at the end of the month is for post-review and post-edit, so you'll need to set aside at least a week for those and preferably more.

**Comment by siakaramalegos** [comment]
@eeeps can you please provide an update here? It doesn't look like there has been any progress so I want to know the plan to get back on track. Thanks

**Comment by eeeps** [comment]
@siakaramalegos 
Apologies, I have been (and am!)

1. behind
2. working a lot locally instead of pushing
3. behind on GH notifications...

Status, with one week to go: query PR is merged; some charts are done; draft is half written. Plan A was to have a fully-written draft (with charts) to turn over to the review/edit folks today but I am going to need at least today to finish the writing. The review/edit folks are aware of the time crunch and have time to devote to this this week, so it seems possible to have a reviewed draft completed by the end of the week.

**Comment by siakaramalegos** [comment]
hi @eeeps where is the draft? I don't see it in the [draft document](https://docs.google.com/document/d/117Roz2RlgXdDl9OiDdkrrd1ahnepFeGzIvTB5S4OY9g/edit#heading=h.l58oy8wsputh).

**Comment by eeeps** [comment]
@siakaramalegos I've been working locally in Markdown. Just copied and pasted over to the draft document. Unfortunately I didn't make as much progress as I wanted to this week and still have a few sections to write, before handing over to the editing team.

**Comment by eeeps** [comment]
@siakaramalegos I did not quite make the progress I wanted to this week, either. I am going to be away from keyboards until Monday morning, but should be able to devote all day Monday to this and am hoping to finally turn things over to @MichaelLewittes and @nhoizey by Tuesday.

**Comment by MichaelLewittes** [comment]
Will look out for it then. Thanks!

**Comment by eeeps** [comment]
@siakaramalegos The draft has now been reviewed by both @nhoizey (technical) and @MichaelLewittes (editorial). I'm starting the process of converting to markdown and creating a PR.

**Comment by eeeps** [comment]
I think the #3143 is ready to merge!

**Comment by svgeesus** [comment]
I noticed this in the [2022 Media chapter](https://almanac.httparchive.org/en/2022/media#image-color-spaces):

> One caveat: AVIF and PNG allow tagging images with wide-gamut color spaces using format-specific shorthands, without using ICC profiles. We started down the path of trying to detect wide-gamut AVIFs and PNGs that don’t use ICC profiles, but accounting for the various ways they are encoded—and the ways our tooling reported on them—proved a bit too complex to tackle this year. Maybe next year!

It was unclear to me, for PNG if you were talking about [`gAMA`](https://w3c.github.io/PNG-spec/#11gAMA) and [`cHRM`](https://w3c.github.io/PNG-spec/#11cHRM) on the one hand or [`cICP`](https://w3c.github.io/PNG-spec/#cICP-chunk) on the other. I would expect **usage for PNG to be low** either way,  `gAMA` and `cHRM` because it is easier and more common to drop an ICC profile into [`iCCP`](https://w3c.github.io/PNG-spec/#11iCCP), and `cICP` because it is so new.

The usage in AVIF is all CICP H.273 metadata, same as PNG `cICP`.

The W3C PNG Working Group, which maintains the PNG specification,  would be interested to help you detect usage of these next year. Is there a pointer to the difficulties that were encountered when trying to do that, this year?

cc/ @eeeps @siakaramalegos @rviscomi

**Comment by eeeps** [comment]
@svgeesus Thank you for reaching out!!

I was not talking about any particular PNG chunk, because all we had to work from was what is reported by `exiftool` and ImageMagick `identify` at some default settings... not the actual PNGs or any ability to parse them ourselves.

For AVIFs,`exiftool` reported the "nclx" string in a field called `Quicktime.ColorRepresentation`, which I made some progress on parsing and mapping to color space names before giving up.

All I could figure out for wide-gamut PNGs (without ICC) were the XY primaries, which turned up in `identify`'s output, under `chromaticity`. I wasn't sure if I could map those to named color spaces.

One piece of example output (for an AVIF), and my half-finished attempts at parsing it, are here: https://observablehq.com/@eeeps/color-space-determiner

If we can figure out a way to reliably turn what we're currently getting from `exiftool`/`identify` into reliable color space categorization, well that's fantastic. I do wonder if making this better in the underlying tooling would be more useful for more people, though (if, of course, much harder).
